src/chat/CustomerSupport/Domain.cs

```diff
@@ -1,13 +1,30 @@
 ﻿using System.Text.Json.Serialization;
 using Microsoft.Extensions.AI;
+using Microsoft.Extensions.VectorData;
+using UglyToad.PdfPig.Fonts.Type1;
+
+internal class EmbeddingDimensions
+{
+    public const int OllamaEmbeddingSize = 384;
+    public const int OpenAIEmbeddingSize = 1536;
+}
 
 public class ManualChunk
 {
+    [VectorStoreRecordData]
     public int ChunkId { get; set; }
+
+    [VectorStoreRecordKey]
     public int ProductId { get; set; }
+
+    [VectorStoreRecordData]
     public int PageNumber { get; set; }
+
+    [VectorStoreRecordData]
     public required string Text { get; set; }
-    public required Embedding<float> Embedding { get; set; }
+
+    [VectorStoreRecordVector(EmbeddingDimensions.OpenAIEmbeddingSize, DistanceFunction.CosineSimilarity)]
```

Comment:
FYI, if you want to use different embedding sizes with the same data model, you can use the separate definition format for describing the metadata instead of attributes.  See [here](https://learn.microsoft.com/en-us/semantic-kernel/concepts/vector-store-connectors/schema-with-record-definition?pivots=programming-language-csharp).